_source/_posts/2021-02-26-unity-webgl-playfab-authorization.md

```diff
@@ -0,0 +1,654 @@
+
+{% img blog/unity-webgl-playfab-authorization/p_Unity_Playfab_Auth.png alt:"PlayFab SDK" width:"800" %}{: .center-image }
+
+## Prerequisites
+
+- Sign up for a free [Azure Playfab account](https://developer.playfab.com/en-US/sign-up).
+- Sign up for a free [Azure account](https://azure.microsoft.com/en-us/free/).
+
+- Download and install [Unity](https://unity.com/).
+- Create a new Unity project from Unity Hub with the 3D template
+- Download and configure the [Playfab Unity SDK and Editor Extensions](https://docs.microsoft.com/en-us/gaming/playfab/sdks/unity3d/quickstart)
+
+## Build Target and Design
+
+For this project, I will be targeting [Unity WebGL](https://docs.unity3d.com/Manual/webgl-gettingstarted.html). The build target is important because different platforms will require different means of facilitating user interaction. 
+
+{% img blog/unity-webgl-playfab-authorization/p_state_model.png alt:"Auth Model" width:"800" %}{: .center-image }
+
+In [Identity Security for Games in C# with Unity](/blog/2020/08/21/unity-csharp-games-security ), I described both native and [OAuth](https://developer.okta.com/blog/2019/10/21/illustrated-guide-to-oauth-and-oidc) design concepts. While building out a user interface for authentication natively in the engine might seem like the best approach because of user experience, it is not the preferred approach for security, and it typically adds much more effort for the developer. This is because it requires the developer to build out logic supporting the entire [authentication state machine](https://developer.okta.com/docs/reference/api/authn/#transaction-state); securely handling every event, every MFA state, registration, MFA enrollment, user self service (account unlock and password reset), etc. Prebuilt widgets can simplify the amount of effort in this regard. However, developers will still be asking a user to enter a password into an untrusted UI that they are responsible for and typically not able to fully keep secure. This is why OAuth is considered the best practice standard for authorization within the security industry, and utilizing a security platform's hosted login experience is considered the most secure way to handle authentication. Players will be entering their credentials into a trusted browser window and a trustworthy security platform. Best of all, developers can share the security responsibility, which is much easier and more secure than attempting to build it themselves.
```

Comment:
I'm having trouble parsing this sentence:

> enter a password into an untrusted UI that they are responsible for and typically not able to fully keep secure

I _think_ you're trying to say this:

> enter a password into an untrusted UI that the game developer is typically not able to control

Reply:
Suggest changing this:

>  Best of all, developers can share the security responsibility, which is much easier and more secure than attempting to build it themselves.

To something like this:

>  Best of all, developers can share the responsibility of keeping the login experience with the security platform, which is much easier and more secure than attempting to do it all themselves.

Reply:
> I'm having trouble parsing this sentence:
> 
> > enter a password into an untrusted UI that they are responsible for and typically not able to fully keep secure
> 
> I _think_ you're trying to say this:
> 
> > enter a password into an untrusted UI that the game developer is typically not able to control

I tried formatting it a bit differently. 

Stole the rest.